kasa/iotprotocol.py

```diff
@@ -51,29 +60,21 @@ async def _query(self, request: str, retry_count: int = 3) -> Dict:
         for retry in range(retry_count + 1):
             try:
                 return await self._execute_query(request, retry)
-            except httpx.CloseError as sdex:
-                await self.close()
```

Comment:
Any specific reason for removing the close calls?

Reply:
When you pass the http session from HA you can't then close the connections and recreate them internally.  However the closing was ported over from the TPLinkSmartHomeProtocol and isn't really necessary for an http library which should be able to re-use the client after errors.

Reply:
That sounds rather odd, are you sure about this? I'm worried that if we don't do resource handling correctly inside the library, it might bite someone else who may not be using homeassistant through leaking sockets or something else.

Reply:
I'm fairly sure but I'll double check

Reply:
I'm pretty sure you're not supposed to `close()` http clients on error.  I've looked through the HA code base and there's not a single example of anyone closing aiohttp or httpx connections after catching exceptions.  Also the way these sessions are shared between integrations would mean closing the client would break all the other integrations. Finally, if you do try to close the client http stops you and [issues a warning](https://github.com/home-assistant/core/blob/7e1dc2286f86060a8fd47cc07ccf2143b9157305/homeassistant/helpers/httpx_client.py#L92).

Reply:
I suppose that works because homeassistant is handling freeing the resources when it deems it's done with them, but wouldn't this leak resources _if_ the client is not passed to us?